lib/cn/node_split.c

```diff
@@ -536,6 +537,7 @@ cn_split(struct cn_compaction_work *w)
         return merr(ENOMEM);
 
     wq = cn_get_io_wq(w->cw_tree->cn);
+    inflight = w->cw_kvset_cnt;
```

Comment:
I didn't know that we can operate on an _Atomic type like its non-atomic counterpart, i.e., without using atomic_load/store/fetch.. interfaces..

Although handy, a downside with this approach is that it reduces code readability as I now need to check the declaration to know whether we are operating on an atomic or not.

Reply:
Yeah, they are effectively intrinsic now, and by default have CST semantics (whereas our default atomic have relaxed semantics IIRC).  I do agree it's not as obvious, and am wafflling on using it.  Perhaps if we required a naming convention on all atomics it would make it more obvious and palatable?

Reply:
Yes, having a naming convention will be certainly helpful!

Reply:
I vote for using the set/get/etc interfaces.

Reply:
I will  need to create a bunch of atomic_XXX_cst() interfaces...

Reply:
Oh.  Well that changes things.  Yuck.  I vote for naming convention.

Reply:
Naming conventions are dumb when you can just hover an identifier to see its declaration using good developer tooling.

Reply:
Atomics should to be highlighted.  I want them to stand out so everybody can easily identify them and give them the extra attention they deserve.

Reply:
I opened a ticket to address this, will get on it right away:
https://jira.micron.com/jira/browse/NFSE-5344